packages/@aws-cdk/aws-stepfunctions-tasks/lib/start-execution.ts

```diff
@@ -39,6 +41,8 @@ export interface StartExecutionProps {
  * A Step Functions Task to call StartExecution on another state machine.
  *
  * It supports three service integration patterns: FIRE_AND_FORGET, SYNC and WAIT_FOR_TASK_TOKEN.
+ *
+ * @deprecated - use 'StepFunctionsStartExecution'
```

Comment:
Same comment as with batch. 

Can we call `new StepFunctionsStartExecution()` in the constructor here and delete most of the code in this file?

Reply:
I think that's pretty natural and organic if we were replacing one API with another to create an adapter and call into the replacement. I don't think this is a scenario where we can just proxy into the replacement as the replacement contains and configures properties (i.e. inputPath, outputPath, ...) that the older 

In this case, we are replacing a class that implemented a bind method to contribute a portion of the properties to represent a Task with a class that's a construct and configures all of the properties for a Task state.

The user experience with the current format will be to embed the call to start execution into the invocation of a new `Task`. I don't think proxying to invoke a new StepFunctionStartExeuction yields the right replacement.

```ts
new Task(this, 'my-task', {
  task: new tasks.StartExecution({...}),
...
}

Some of the other ones to come like DynamoDB and ECS, I've tried to reuse the shared types. I could perhaps move more into a central location here and DRY it up a bit.

Did you have something else in mind? If so, are you opposed to me making a subsequent PR to do some refactoring?
```